File: app/src/main/java/me/tatarka/bindingcollectionadapter/sample/FragmentSpinnerView.java
package me.tatarka.bindingcollectionadapter.sample;

import androidx.lifecycle.ViewModelProviders;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import me.tatarka.bindingcollectionadapter.sample.databinding.SpinnerViewBinding;

/**
 * Created by evan on 5/31/15.
 */
public class FragmentSpinnerView extends Fragment {
    private static final String TAG = "BindingSpinner";
    private MutableViewModel viewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = ViewModelProviders.of(this).get(MutableViewModel.class);
        viewModel.setCheckable(false);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        SpinnerViewBinding binding = SpinnerViewBinding.inflate(inflater, container, false);
        binding.setLifecycleOwner(this);
        binding.setViewModel(viewModel);
        binding.setListeners(viewModel);
        binding.executePendingBindings();
        return binding.getRoot();
    }
}

File: app/src/main/java/me/tatarka/bindingcollectionadapter/sample/FragmentSpinnerView.kt
package me.tatarka.bindingcollectionadapter.sample

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.ViewModelProvider

import me.tatarka.bindingcollectionadapter.sample.databinding.SpinnerViewBinding

class FragmentSpinnerView : Fragment() {
    private lateinit var viewModel: MutableViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProvider(this).get(MutableViewModel::class.java)
        viewModel.setCheckable(false)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return SpinnerViewBinding.inflate(inflater, container, false).also {
            it.lifecycleOwner = this
            it.viewModel = viewModel
            it.listeners = viewModel
            it.executePendingBindings()
        }.root
    }
}
